joblib/numpy_pickle_utils.py

```diff
@@ -147,20 +151,42 @@ def _read_fileobject(fileobj, filename, mmap_mode=None):
     """
     # Detect if the fileobj contains compressed data.
     compressor = _detect_compressor(fileobj)
+    if isinstance(fileobj, tuple(_COMPRESSOR_OBJS)):
```

Comment:
I think my question was why you were using yield rather than return but this discussion is probably best left out of this PR.